packages/twenty-front/src/modules/object-record/select/hooks/useRecordsForSelect.ts

```diff
@@ -80,42 +80,26 @@ export const useRecordsForSelect = ({
     })
     .filter(isDefined);
 
+  const filteredSelectedRecordFilter = { id: { in: selectedIds } };
   const {
     loading: filteredSelectedRecordsLoading,
     records: filteredSelectedRecordsData,
   } = useFindManyRecords({
-    filter: {
-      and: [
-        {
-          and: searchFilter,
-        },
-        {
-          id: {
-            in: selectedIds,
-          },
-        },
-      ],
-    },
+    filter: searchFilters.length
```

Comment:
The previous way was sometimes duplicating cache entries.

For instance, on main branch, here are three entries I'm seeing in the Apollo cache:

<img width="849" alt="image" src="https://github.com/twentyhq/twenty/assets/3098428/e91326cc-d3e4-4467-bf41-188554a08f12">

<img width="726" alt="image" src="https://github.com/twentyhq/twenty/assets/3098428/65cba008-9e99-4776-bed4-6d5fe8d8c345">

Those three entries should be one unique entry as they in fact represent the same query result. But Apollo does not know this. It uses the variables to identify the cache entries, and in this case the variables are different because of the empty "and" and "not" filters, which is causing Apollo to store one entry for each of the variables variant.

This is why I am only adding the "and" if necessary: if filters are always written the same way, we can optimize the cache, and therefore the optimistic effects will be optimized as well as they will be triggered on less cache entries.

For this example, ideally we wouldn't even use a `filter` variable as the filters are not filtering anything. These three cache entries would be merged as one:
```opportunities({ "first": 60, "orderBy": { "createdAt": "AscNullLast" } })```

As you suggested above, I'll try to implement an util to simplify filters and use it here instead.

Reply:
Done, I added two filter utils: `andFilterVariables` and `orFilterVariables`. They simplify the filter so it only uses an "and" or "or" when necessary.

I added checking for some `not: { id: { in: [/* array of ids */] } }` filters that the array of ids is not empty.

And finally, I also added some "skip" params for `useFindManyRecords` for the entity search when there are no selected ids, so we don't fetch selected records if we already know there are none selected!

With this I managed to only have this in the cache instead of the three entries I mentioned:

<img width="552" alt="image" src="https://github.com/twentyhq/twenty/assets/3098428/bae5eeda-c9ba-4ae5-8691-0436cabce21c">

When I select an opportunity in the Entity Search Select, the following entries are added to the cache:

<img width="565" alt="image" src="https://github.com/twentyhq/twenty/assets/3098428/11548b6e-98b5-45c6-824b-21ed643dc41f">

So we're only adding filtered query entries when needed.

Reply:
Just noticed in the last entry, the "not" has twice the same id, I believe this is because of this piece of code in `RecordRelationFieldCardSection`:

```
const entities = useFilteredSearchEntityQuery({
    filters: [
      {
        fieldNames:
          searchQuery?.computeFilterFields?.(
            relationObjectMetadataNameSingular,
          ) ?? [],
        filter: relationPickerSearchFilter,
      },
    ],
    orderByField: 'createdAt',
    selectedIds: relationRecordIds,
    excludeEntityIds: relationRecordIds, // => Here
    objectNameSingular: relationObjectMetadataNameSingular,
  });
  ```
I removed the `excludeEntityIds`, seems to fix the issue!